### PR TITLE
[IT-1766] Fix and add service control policies

### PR DESCRIPTION
**Pull request by zaro0508**
* Fix SCP to restrict AWS regions from being used.
* Add a policy to not allow admins of an account to leave an organization.

### CODE REVIEW

File: org-formation/_scp.yaml [hunk 1]
```diff
@@ -6,21 +7,55 @@
...
-              - 'support:*'
-            Resource: '*'
-            Condition:
-              StringNotEquals:
-                'aws:RequestedRegion':
-                  - us-west-1
-                  - us-west-2
-                  - us-east-1
-                  - us-east-2
+        - Sid: DenyAllOutsideEU
```
Comment (xschildw) [inline org-formation/_scp.yaml:10]: US?

File: org-formation/_scp.yaml [hunk 2]
```diff
@@ -126,3 +161,19 @@
...
+  # from https://docs.aws.amazon.com/organizations/latest/userguide/orgs_manage_policies_scps_examples_general.html#example-scp-deny-region
+  RestrictLeaveOrganizationSCP:
+    Type: OC::ORG::ServiceControlPolicy
+    Properties:
+      PolicyName: RestrictLeaveOrganization
+      Description: Administrators of member accounts can't remove their accounts from the organization
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: DenyDnsRegistration
```
Comment (xschildw) [inline org-formation/_scp.yaml:174]: Could you confirm? That does not match the description and we need to be able to register / transfer domains (or are you planning to remove the permissions in all accounts except sageit?).

File: org-formation/_scp.yaml (hunk 1 above)
Comment (zaro0508) [reply org-formation/_scp.yaml:10]: good catch

File: org-formation/_scp.yaml (hunk 2 above)
Comment (zaro0508) [reply org-formation/_scp.yaml:174]: oops, did copy/paste without updating description

File: org-formation/_scp.yaml (hunk 1 above)
Comment (zaro0508) [reply org-formation/_scp.yaml]: fixed

File: org-formation/_scp.yaml (hunk 2 above)
Comment (zaro0508) [reply org-formation/_scp.yaml]: fixed

File: org-formation/_scp.yaml [hunk 3]
```diff
@@ -126,3 +161,19 @@
...
+    Properties:
+      PolicyName: RestrictLeaveOrganization
+      Description: Administrators of member accounts can't remove their accounts from the organization
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: DenyLeaveOrganization
+            Effect: Deny
+            Action:
+              - 'route53domains:RegisterDomain'
```
Comment (xschildw) [inline org-formation/_scp.yaml:177]: Should this not be 'organizations:LeaveOrganization'  ?

Comment (zaro0508) [reply org-formation/_scp.yaml]: yes, of course.  fixed!

**Comment by zaro0508** [comment]
could you please take another look @xschildw ?

File: org-formation/_scp.yaml [hunk 4]
```diff
@@ -126,3 +161,18 @@
               - 'route53domains:RegisterDomain'
               - 'route53domains:TransferDomain'
             Resource: '*'
```
Comment (xschildw) [inline org-formation/_scp.yaml:163]: Maybe I'm missing something... There are still some domains that need to be transferred to sageIT, and we need to register domains in sageIT. Does not this rule prohibit this? Do we need a condition here to give admins (or some role) the right to do register/transfer domains?

**Review by xschildw — APPROVED** [review]
Looks good, see my question re: register/transfer domain.

File: org-formation/_scp.yaml (hunk 4 above)
Comment (zaro0508) [reply org-formation/_scp.yaml:163]: yes, your understanding of how this SCP works is correct @xschildw.  However you also need to consider how it's used.  The application of this SCP is in [organizations.yaml](https://github.com/Sage-Bionetworks-IT/organizations-infra/blob/master/org-formation/organization.yaml).  We are applying this to nearly all of our accounts except the `SageITAccount`.

**Pull request merged**